clang/lib/Sema/SemaOverload.cpp

```diff
@@ -11322,6 +11340,15 @@ OverloadingResult OverloadCandidateSet::BestViableFunctionImpl(
   if (S.getLangOpts().CUDA)
     CudaExcludeWrongSideCandidates(S, Candidates);
 
+  // In SYCL host compilation remove candidates marked SYCLDeviceOnly.
+  if (S.getLangOpts().isSYCL() && !S.getLangOpts().SYCLIsDevice) {
```

Comment:
We appear to have `LangOpts.SYCLIsHost` as it's used in this PR above. Could we use that here?